python/dask_cudf/dask_cudf/tests/test_join.py

```diff
@@ -386,3 +386,15 @@ def test_issue_12773():
         expected.to_pandas(),
         check_index=False,
     )
+
+
+@pytest.mark.parametrize(
+    "typ", [cudf.Decimal32Dtype, cudf.Decimal64Dtype, cudf.Decimal128Dtype]
+)
+def test_merge_on_decimal(typ):
+    df = cudf.DataFrame({"a": [1], "b": [2]}, dtype=typ(1))
+    dask_frame = dask_cudf.from_cudf(df)
+    result = dask_frame.merge(dask_frame, left_on="a", right_on="a").compute()
+    expected = df.merge(df, left_on="a", right_on="a")
+    dd.assert_eq(result.to_pandas(), expected.to_pandas())
```

Comment:
Why the `to_pandas` conversion? What is failing with the cudf? Also, do you need the compute in

```python
result = dask_frame.merge(dask_frame, left_on="a", right_on="a").compute()
```

In my mind I imagine this test looking something like:
```python
    df = cudf.DataFrame({"a": [1], "b": [2]}, dtype=typ(1))
    ddf = dd.from_pandas(df)
    result = ddf.merge(ddf, left_on="a", right_on="a")
    expected = df.merge(df, left_on="a", right_on="a")
    dd.assert_eq(result, expected)
```
I'm curious what fails in `dd.assert_eq(result, expected)`?

Reply:
Ah thanks for this simplification. I was just ignorantly mirroring the test above this one which used `to_pandas` and `compute`. This test works with your simplification